mypy/main.py

```diff
@@ -492,6 +492,10 @@ def add_invertible_flag(flag: str,
                         help="Treat imports as private unless aliased",
                         group=strictness_group)
 
+    add_invertible_flag('--disallow-inferred-unreachable', default=False, strict_flag=False,
```

Comment:
Maybe we can make the flag name a bit clearer. What about `--warn-unreachable`? We still allow some unreachable code, but I think that we don't need to make this distinction in the flag name. I think that it's sufficient to describe the semantics in the documentation.